### PR TITLE
[master] file.keyvalue should allow creating a file if it doesn't exist

### LINKED ISSUE #63545: file.keyvalue should create the file if it doesn't exist
If you don't yet have a file that you would like to create by key/value pairs, then this state should just create it.
Or at least be able to do it by using a "create=xxx" option.

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Adds the `create_if_missing` parameter to handle basic file creation for `file.keyvalue`

### What issues does this PR fix or reference?
Fixes: #63545

### Previous Behavior
`file.keyvalue` could not create a file, so `file.touch` or `file.managed` without content were often used.

### New Behavior
Basic file creation is now possible, although management of permissions and ownership is not.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/states/file.py [hunk 1]
```diff
@@ -5564,7 +5570,16 @@ def keyvalue(
     try:
         with salt.utils.files.fopen(name, "r") as fd:
             file_contents = fd.readlines()
-    except OSError:
+            fd.close()
```
Comment (Ch3LL) [inline salt/states/file.py:5573]: Shouldn't the context manager close this file properly? Why are you adding the close call here?

Comment (nicholasmhughes) [reply salt/states/file.py]: I just got caught up in the magnificent chaos of it all.

**Review by Ch3LL — DISMISSED** [review]

**Comment by nicholasmhughes** [comment]
@Ch3LL ... oooooo ... we get f-strings now? do you need me to run pre-commit and commit the new formatting changes?

**Comment by Ch3LL** [comment]
Yeah if you don't mind. Recently a change was made to the pre commit config file to make the code compatible with 3.8 and up, since 3.7 is going end of life in a few weeks.

**Review by MKLeb — DISMISSED** [review]

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
@nicholasmhughes sorry but there seems to be a merge conflict. Mind resolving that?

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**